### PR TITLE
Endpoint: Remove endpoint.mitigated field

### PR DESCRIPTION
**Pull request by kiblik**
Remove leftover `endpoint.mitigated` field. Any endpoint that is mitigated will get their `endpoint_statuses` mitigated as part of this PR via a migration.

Part of the solution of https://github.com/DefectDojo/django-DefectDojo/issues/4198

### CODE REVIEW

**Review by damiencarol — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]
This looks good to me, but I am nervous to merge the day before a release. @valentijnscholten leaving this to you if think it's okay to merge

File: dojo/endpoint/views.py [hunk 1]
```diff
@@ -397,14 +397,21 @@ def endpoint_bulk_update_all(request, pid=None):
                 if skipped_endpoint_count > 0:
                     add_error_message_to_response('Skipped mitigation of {} endpoints because you are not authorized.'.format(skipped_endpoint_count))
 
+                eps_count = 0
+
                 for endpoint in endpoints:
-                    endpoint.mitigated = not endpoint.mitigated
-                    endpoint.save()
+                    for eps in endpoint.endpoint_status:
```
Comment (valentijnscholten) [inline dojo/endpoint/views.py:403]: might be good to prefetch the endpont_status relation to make it performant on large number of endpoints?

**Comment by valentijnscholten** [comment]
If it's obsolete, why does it need a migration? Could it lead to surprises with endpoint_statuses being closed?

**Comment by kiblik** [comment]
> If it's obsolete, why does it need a migration? Could it lead to surprises with endpoint_statuses being closed?

If somebody marked endpoint as mitigated (by GUI or API), I suppose his action/decision should be migrated.
On the other hand, mitigation should be changed by (re)importer, not by a user (if he/she is not doing manual pentest) and he/she should use flags "false positive", "out of scope" and "risk accepted"

File: dojo/endpoint/views.py (hunk 1 above)
Comment (kiblik) [reply dojo/endpoint/views.py:403]: Hmmm... and maybe use of [`.update()`](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.update) would be the best idea

**Comment by valentijnscholten** [comment]
Looking at the PR it seems the endpoint.mitigated field is no longer shown in the UI? This migration would then use the value of this "hidden/no longer visible" field to start mitigating endpoint_statuses. This might be confusing for some users? Not sure, just thinking out loud. In theory the affected endpoint statuses should be small.

**Comment by valentijnscholten** [comment]
I don't know the history of changes around this field to make a good judgement. I think impact will be small so might as well go for it.

File: dojo/endpoint/views.py (hunk 1 above)
Comment (valentijnscholten) [reply dojo/endpoint/views.py:403]: @kiblik are you planning on changing this so we can merge it?

Comment (kiblik) [reply dojo/endpoint/views.py:403]: I will prepare it today/tomorrow.

**Comment by kiblik** [comment]
The last requested change was implemented. Feel free to merge it.

**Pull request merged**